### PR TITLE
Fix #247: Use https links instead of http

### LINKED ISSUE #247: Use HTTPS links
Hello,
this project uses multiple HTTP instead of HTTPS links in markdown files, Java documentation (and exception messages) and POM files.
It would be good to replace those with HTTPS since most target sites, to which the links refer, support it.

I am not going to sign the CLA or submit a pull request for this, so if you don't consider this a problem then feel free to close this issue.
However, in case you do consider this an issue, here is a regex pattern for finding such links (excluding some false positives):
```
http://(?!maven\.apache\.org/POM|www\.w3\.org/2001/XMLSchema-instance|maven\.apache\.org/maven-.+\.xsd|www\.eclipse\.org/legal/epl-v10\.html|www\.eclipse\.org/org/documents/edl-v10\.php|www\.w3\.org/TR/xhtml1/DTD/.+|www\.w3\.org/1999/xhtml|maven\.apache\.org/SETTINGS/.+|maven\.apache\.org/xsd/.+|(www\.)?example\.com/.*)
```

Additionally there are tools such as Spring's [nohttp](https://github.com/spring-io/nohttp) which can automate finding such links in the future (though I have not used it myself yet).

### PR DESCRIPTION
**Pull request by mschreiber**
All links to documentation, schemas, etc. are now using https instead of http

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -156,7 +156,7 @@ To use the profiler, [set the system property](https://github.com/jcgay/maven-pr
 
 To get started with YouMonitor, you need to install and run the application. It will ask you for a repository, which is how you aggregate builds (e.g. use one repository per different project that you want to investigate). Afterwards select the [Monitoring in IDE or command line](https://www.yourkit.com/docs/youmonitor/help/ide_and_command_line.jsp) and use the button "Open Instructions". That will show you the project and machine specific argument which needs to be added to the Java command line. E.g. if you want to profile tests, you might want to add it to the [argLine configuration of Tycho Surefire](https://www.eclipse.org/tycho/sitedocs/tycho-surefire-plugin/test-mojo.html#argLine).
 
-## 🏗️ Build & Test
+## �?��? Build & Test
```
Comment (akurtakov) [inline CONTRIBUTING.md:160]: Broken emoji.

File: demo/itp04-rcp/example-feature/epl-v10.html [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <html xmlns:o="urn:schemas-microsoft-com:office:office"
 xmlns:w="urn:schemas-microsoft-com:office:word"
-xmlns="http://www.w3.org/TR/REC-html40">
+xmlns="https://www.w3.org/TR/REC-html40">
```
Comment (akurtakov) [inline demo/itp04-rcp/example-feature/epl-v10.html:3]: xmlns shouldn't be changed as it's an identifier.

**Review by akurtakov — COMMENTED** [review]
Please check other instances for similar issues. Overall, it's too big to be practically reviewable.

**Comment by mschreiber** [comment]
I will create different smaller pull requests....

**Pull request closed without merging**